avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/ArrayVisitor.java

```diff
@@ -35,8 +37,8 @@ public Schema builtAvroSchema() {
             throw new IllegalStateException("No element schema created for: "+_type);
         }
         Schema schema = Schema.createArray(_elementSchema);
-        if (_type.isArrayType()) {
-            schema.addProp(AVRO_SCHEMA_PROP_CLASS, _type.toCanonical());
+        if (!_type.hasRawClass(List.class)) {
+            schema.addProp(AVRO_SCHEMA_PROP_CLASS, AvroSchemaHelper.getTypeId(_type));
```

Comment:
And to be a bit more specific, if Scala's `List` equivalent does not at least implement `java.util.Collection`, then it is not compatible with Avro's `ARRAY` type when using the Apache implementation, and would be serialized as a record.